### ISSUE TITLE
Since some weeks tasks in my NB development environment do not end any more.

### ISSUE THREAD

**Issue by WernerSchulte** [apache/netbeans#5147]
### Apache NetBeans version

Apache NetBeans 16

### What happened

Since some weeks tasks in my NB development environment do not end any more.
Instead the status line shows e.g. '12 more'. If i click that message the tasks (debugger, clean build, etc) are shown with the indeterminate progress bar. A click in the 'cancel' button doesnt help. Instead closing NB that list of tasks is presented to the user to be finished. After confirm, NB shuts down and the tasklist is shown again for confirmation

This was observed in NB 12.5 which I used for years and never had such problem.
I update the NB16 (and took over the configuration) - Same problem

I was checking the IDE Log and found a message
java.lang.IllegalArgumentException: Key contains code point U+0000

Notification says, there is an unexpected Exception (each time  clean build or debugger or run was finished
[messages.log](https://github.com/apache/netbeans/files/10301078/messages.log)

### How to reproduce

I guess this is a 'broken' configuration issue - No way to reproduce ... 
I have a lot of projects, which I am working on frequently.
The option to remove the 'Roaming/Netbeans' folder would be most annoying, because I would need to setup the configuration full again (will be a pain)

### Did this work correctly in an earlier version?

Apache NetBeans 12.6 or earlier

### Operating System

Windows 10

### JDK

OpenJDK 15

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

Happens for each task done in the netbeans 

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by WernerSchulte** [comment]
[uigestures.txt](https://github.com/apache/netbeans/files/10301123/uigestures.txt)
Found the U+0000 message in here

**Comment by WernerSchulte** [comment]
![image](https://user-images.githubusercontent.com/121410858/209534165-abfcf444-ee09-4a37-a274-18a2654c686e.png)

After a day of working it looks like that (when closing NB)

**Comment by lkishalmi** [comment]
You have to clean up your user/cache dirs. There is no way around that.
The U+0000 happens due to this one: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8075156

**Comment by WernerSchulte** [comment]
Can I at least somehow extract / restore the ProjectGroups configurations ?

That problem occurred all over sudden and might occur again in the future.
I don't want to recreate all groups again and again

Netbeans for me is clearly the best Java development tool.
But this habit makes it hard to continue developing in a efficient way

**Comment by lkishalmi** [comment]
You might take a chance saving/restoring the content of `$USER_DIR/config/Preferences/org/netbeans/modules/projectui` directory.

**Comment by WernerSchulte** [comment]
[JFindNBPrefsNulls.zip](https://github.com/apache/netbeans/files/10311501/JFindNBPrefsNulls.zip)
Implemented a Prefs Checker for NB Prefs folder (in java - see attachment).
With this one I found, the uihandler.properties had a huge bunch of \u0000 (String)
and customDefs.properties had real binary 0x00 char.

I first removed the uihandler.properties and restarted - same problem, same message with the u0000 codepoint
Then I removed the customDefs.properties and netbeans was happy (started tasks and properly).

I guess this happens, when somehow Windows FS is broken or a 'switch off' is required when netbeans is trying to store the prefs.

Take the code and maybe improve NB to find a broken prefs instead of letting the user getting mad when unexpected exceptions happen and its unlear, what is the outcome ...
